tools/test.py

```diff
@@ -138,6 +142,20 @@
             # Print available tests in order and exit
             print_tests(tests, options.format)
             sys.exit(0)
+        elif options.export:
+            base_source_paths = options.source_dir
+            # Default base source path is the current directory
+            if not base_source_paths:
+                base_source_paths = ['.']
+            target = options.mcu
+            for test_name, test_path in tests.iteritems():
+                src_path = base_source_paths + [test_path]
+                # don't clean, no zip, using sources
+                tmp_path, report = export(src_path, test_name, options.ide, target, src_path[0], tempdir=options.build_dir, clean=False, make_zip=False, sources_relative=True if base_source_paths else False)
```

Comment:
At some point before this call it'd be good to check if `options.ide` was actually specified and error accordingly.